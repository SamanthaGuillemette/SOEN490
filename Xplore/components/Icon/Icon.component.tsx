import { Feather } from "@expo/vector-icons";
import { StyleProp, ViewStyle } from "react-native";
import { colors } from "../../constants";
import { useThemeColor } from "../../hooks";

interface IconProps {
  name: keyof typeof Feather.glyphMap;
  size?: "small" | "medium" | "large";
  color: keyof typeof colors.light & keyof typeof colors.dark;
  style?: StyleProp<ViewStyle>;
}

export const Icon = ({ name, size = "large", color, style }: IconProps) => {
  const iconColor = useThemeColor(color);

  let iconSize: number = 24;
  switch (size) {
    case "small":
      iconSize = 15;
      break;
    case "medium":
      iconSize = 18;
      break;
    case "large":
      iconSize = 24;
      break;
  }

  return (
    <Feather name={name} size={iconSize} color={iconColor} style={style} />
  );
};
