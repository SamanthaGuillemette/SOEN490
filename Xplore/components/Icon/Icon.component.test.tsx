import React from "react";
import Icon from "./Icon.component";
import renderer from "react-test-renderer";
import { cleanup } from "react-native-testing-library";

afterAll(() => {
  cleanup();
});

describe("Icon component", () => {
  it("should render icon correctly", () => {
    const icon = renderer.create(
      <Icon name={"award"} color="primary" />
    ).toJSON;
    expect(icon).toMatchSnapshot();
  });
});
