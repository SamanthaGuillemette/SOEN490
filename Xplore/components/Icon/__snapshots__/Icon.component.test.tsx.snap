// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render icon correctly 1`] = `[Function]`;
